superset/db_engine_specs/trino.py

```diff
@@ -127,14 +129,42 @@ def get_tracking_url(cls, cursor: "Cursor") -> Optional[str]:
         return None
 
     @classmethod
-    def handle_cursor(cls, cursor: "Cursor", query: Query, session: Session) -> None:
-        """Updates progress information"""
+    def handle_cursor(cls, cursor: Cursor, query: Query, session: Session) -> None:
         tracking_url = cls.get_tracking_url(cursor)
         if tracking_url:
             query.tracking_url = tracking_url
-            session.commit()
+
+        # Adds the executed query id to the extra payload so the query can be cancelled
+        query.set_extra_json_key("cancel_query", cursor.stats["queryId"])
+
+        session.commit()
         BaseEngineSpec.handle_cursor(cursor=cursor, query=query, session=session)
 
+    @classmethod
+    def has_implicit_cancel(cls) -> bool:
+        return False
```

Comment:
I feel some of these class methods in DBEngineSpec should probably be attributes instead. But that's another story.

Reply:
Yeah, I was annoyed by the same detail, but then remembered you can't place a `@property` on a `@classmethod` 🙂 I think it would almost make more sense if these were instantiated, as it always feels slightly strange that these are uninstantiated classes.